core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3810,16 +3811,27 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
-  // Just a placeholder for now.
   def handleListClientMetricsResources(request: RequestChannel.Request): Unit = {
     val listClientMetricsResourcesRequest = request.body[ListClientMetricsResourcesRequest]
 
     if (!authHelper.authorize(request.context, DESCRIBE_CONFIGS, CLUSTER, CLUSTER_NAME)) {
       requestHelper.sendMaybeThrottle(request, listClientMetricsResourcesRequest.getErrorResponse(Errors.CLUSTER_AUTHORIZATION_FAILED.exception))
     } else {
-      // Just return an empty list in the placeholder
-      val data = new ListClientMetricsResourcesResponseData()
-      requestHelper.sendMaybeThrottle(request, new ListClientMetricsResourcesResponse(data))
+      clientMetricsManager match {
+        case Some(metricsManager) =>
+          try {
+            val data = new ListClientMetricsResourcesResponseData().setClientMetricsResources(
+              metricsManager.listClientMetricsResources.asScala.map(
+                name => new ClientMetricsResource().setName(name)).toList.asJava)
+            requestHelper.sendMaybeThrottle(request, new ListClientMetricsResourcesResponse(data))
+          } catch {
+            case _: Exception =>
```

Comment:
Do we need this? `KafkaApis` has a generic way to handle unexpected errors through `handleError`.

Reply:
Yeah, this is not needed in this API. I have removed it and validated in test that API returns `UNKNOWN_SERVER_ERROR` now.

Reply:
Done.